transport/src/main/java/io/netty/channel/ChannelOutboundBuffer.java

```diff
@@ -828,4 +847,49 @@ Entry recycleAndGetNext() {
             return next;
         }
     }
+
+    // the cumulator copied from ByteToMessageDecoder
```

Comment:
if this is duplicated code, can we share it? Maybe make something available in the `netty-buffer` package @normanmaurer?